pytest.ini

```diff
@@ -1,8 +1,15 @@
 [pytest]
-addopts=--doctest-modules --doctest-glob=pkg_resources/api_tests.txt -r sxX
 norecursedirs=dist build .tox .eggs
-doctest_optionflags=ELLIPSIS ALLOW_UNICODE
-filterwarnings =
+addopts=
+	--doctest-modules
+	--doctest-glob=pkg_resources/api_tests.txt
+	-r sxX
+doctest_optionflags=ALLOW_UNICODE ELLIPSIS
+# workaround for warning pytest-dev/pytest#6178
+junit_family=xunit2
+filterwarnings=
+    # https://github.com/pytest-dev/pytest/issues/6928
+    ignore:direct construction of .*Item has been deprecated:DeprecationWarning
```

Comment:
Shouldn't ignores go after the `error` line? I think the order matters.

Reply:
The "fail on warnings" behavior is project-specific whereas the 'ignore "direct construction of..."' is part of the upstream skeleton. Interleaving these changes would make reconciling changes more difficult.

I tried to understand if or how order matters, so I looked at [the docs](https://docs.python.org/3/library/warnings.html#the-warnings-filter), which say

> any specific warning is matched against each filter specification in the list in turn until a match is found

That suggests that putting `error` at the front would cause all warnings to become errors and no ignores would take effect. Obviously, that's not what's happening, so it seems we can't trust the docs.

So, I'm not going to worry about it too much, especially since that upstream filter can (and will) likely  be removed soon if not today.

Reply:
It sounds like you're right. Although I'm a bit confused that the docs imply something different.